tests/core/asyncio/test_subscription_wait_apis.py

```diff
@@ -0,0 +1,69 @@
+import asyncio
+
+import pytest
+
+from helpers import DummyRequestPair, DummyResponse
+
+
+@pytest.mark.asyncio
+async def test_wait_until_any_all_subscribed(triplet_of_endpoints):
```

Comment:
@pipermerriam did you see these tests? I think they cover all the new APIs.

Reply:
I did, though I didn't see that they hit all the APIs.  However, I'm inclined to request a test per unit of API functionality.  Under this structure we avoid a future situation where a refactor accidentally removes coverage of an API.  With very granular tests that each try to test a single thing, it becomes much easier to understand what each test is intended to test.

Reply:
Moved them into separate tests :+1: